modules/gui/warp-list.lua

```diff
@@ -73,87 +81,200 @@ Gui.element{
-    sprite = 'utility/rename_icon_normal',
-    tooltip = {'warp-list.edit-tooltip-none'},
-    style = 'tool_button'
-}
-:style(Styles.sprite20)
-:on_click(function(player, element)
-    local warp_id = element.parent.name:sub(6)
-    Warps.set_editing(warp_id, player.name, true)
+    -- Reset the warp cooldown if the player does not have unlimited warps
+    if not check_player_permissions(player, 'bypass_warp_cooldown') then
+        PlayerCooldown:set(player, config.cooldown_duration)
+    end
+
+    PlayerInRange:set(player, warp_id)
 end)
 
---- Set of three elements which make up each row of the warp table
--- @element add_warp_base
-local add_warp_base =
-Gui.element(function(_, parent, warp_id)
-    -- Add the icon flow
-    local icon_flow =
-    parent.add{
-        name = 'icon-'..warp_id,
-        type = 'flow',
-        caption = warp_id
+--- The button that is visible when the warp is in edit state
+-- @element warp_icon_editing
+local warp_icon_editing =
+Gui.element(function(event_trigger, parent, warp)
+    return parent.add{
+        name = event_trigger,
+        type = 'choose-elem-button',
+        elem_type = 'signal',
+        signal = {type = warp.icon.type, name = warp.icon.name},
+        tooltip = {'warp-list.goto-edit'}
     }
-    icon_flow.style.padding = 0
+end)
+:style(Styles.sprite32)
 
-    -- Add a flow which will contain the warp name and edit buttons
-    local warp_flow = parent.add{ type = 'flow', name = warp_id }
-    warp_flow.style.padding = 0
+--- Warp label, visible if the player is not in edit state
+-- @element warp_label
+local warp_label =
+Gui.element(function(event_trigger, parent, warp)
+    local last_edit_name = warp.last_edit_name
+    local last_edit_time = warp.last_edit_time
+    -- Draw the element
+    return parent.add{
+        name = event_trigger,
+        type = 'label',
+        caption = warp.name,
+        tooltip = {'warp-list.last-edit', last_edit_name, format_time(last_edit_time)}
+    }
+end)
+:style{
+    single_line = true,
+    left_padding = 2,
+    right_padding = 2,
+    horizontally_stretchable = true
+}
+:on_click(function(player, element, _)
+    local warp_id = element.parent.caption
+    local warp = Warps.get_warp(warp_id)
+    local position = warp.position
+    player.zoom_to_world(position, 1.5)
+end)
 
-    -- Add the two edit buttons outside the warp flow
-    local edit_flow = Gui.alignment(parent, 'edit-'..warp_id)
-    edit_warp(edit_flow)
-    discard_warp(edit_flow)
+--- Warp status, visible if the player is not in edit state
+--- This will show if the warp is connected or not
+-- @element warp_status
+local warp_status =
+Gui.element(function(event_trigger, parent)
+    -- Draw the element
+    return parent.add{
+        name = event_trigger,
+        type = 'label',
+        caption = '[img=utility/electricity_icon_unplugged]', -- Temporary icon
+    }
+end)
+:style{
+    -- when editing mode because textbox is larger the icon would move up.
```

Comment:
```suggestion
    -- When editing mode because textbox is larger the icon would move up.
```
Generally comments are being capitalised?